File: RealEstate.Domain/Roles/Appoiment.cs
using RealEstate.Domain.Entity;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RealEstate.Domain.Roles
{
    public class Appointment
    {
        public int Id { get; set; }
        public string? Description { get; private set; }
        public DateTime StartTime { get; private set; }
        public DateTime EndTime { get; private set; }
        public AppointmentStatus Status { get; private set; }

    }
    public enum AppointmentStatus
    {
        Pending,
        Confirmed,
        Completed,
        Cancelled
    }
}

File: RealEstate.Presentation/RealEstate/src/app/models/roles/appoiment/appoiment.ts
export enum AppointmentStatus {
    Pending = 'Pending',
    Confirmed = 'Confirmed',
    Completed = 'Completed',
    Cancelled = 'Cancelled'
  }
  
  export class Appointment {
    id: number;
    description?: string;
    startTime: Date;
    endTime: Date;
    status: AppointmentStatus;
  
    constructor(
      id: number,
      startTime: Date,
      endTime: Date,
      status: AppointmentStatus = AppointmentStatus.Pending, 
      description?: string
    ) {
      this.id = id;
      this.startTime = startTime;
      this.endTime = endTime;
      this.status = status;
      this.description = description;
    }
  }
  